File: sms/frontend/src/services/eventService.js
import api from './api'

const BASE = '/api/v1/events'

export const eventService = {
  getAll:       (societyId, params) => api.get(BASE, { params: { societyId, ...params } }),
  getById:      (id)                => api.get(`${BASE}/${id}`),
  getUpcoming:  (societyId)         => api.get(`${BASE}/upcoming`, { params: { societyId } }),
  create:       (data)              => api.post(BASE, data),
  update:       (id, data)          => api.put(`${BASE}/${id}`, data),
  delete:       (id)                => api.delete(`${BASE}/${id}`),
  register:     (id)                => api.post(`${BASE}/${id}/register`),
  unregister:   (id)                => api.delete(`${BASE}/${id}/register`),
}

File: sms/backend/src/main/java/com/society/management/service/EventService.java
package com.society.management.service;

import com.society.management.dto.request.EventRequest;
import com.society.management.dto.response.EventDto;
import com.society.management.dto.response.UserDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Event management operations.
 */
public interface EventService {

    EventDto create(Long organiserId, EventRequest request);

    EventDto update(Long id, EventRequest request);

    EventDto getById(Long id);

    Page<EventDto> getBySociety(Long societyId, Pageable pageable);

    List<EventDto> getUpcomingEvents(Long societyId);

    void delete(Long id);

    void registerUser(Long eventId, Long userId);

    void cancelRegistration(Long eventId, Long userId);

    List<UserDto> getParticipants(Long eventId);
}
